packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -147,6 +149,53 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
         e.preventDefault();
         const newNavigatedDate = navigateToNewDate(gridNavigatedDate, 'Day', -1, restrictedDatesOptions);
 
+        if (!!newNavigatedDate) {
+          setShouldFocusInDayGrid(true);
+          setGridNavigatedDate(newNavigatedDate);
+        }
+      },
+      moveToStartOfWeek: e => {
+        e.preventDefault();
+        const targetDate = getStartDateOfWeek(gridNavigatedDate, firstDayOfWeek);
+        const newNavigatedDate = contstraintNavigatedDate(gridNavigatedDate, targetDate, -1, restrictedDatesOptions);
+
+        if (!!newNavigatedDate) {
+          setShouldFocusInDayGrid(true);
+          setGridNavigatedDate(newNavigatedDate);
+        }
+      },
+      moveToEndOfWeek: e => {
+        e.preventDefault();
+        const targetDate = getEndDateOfWeek(gridNavigatedDate, firstDayOfWeek);
```

Comment:
Continuing discussion from https://github.com/microsoft/fluentui/pull/14862#discussion_r482913630, since PR was split

Reply:
Agreed with @rymeskar offline to keep current implementation like it is